File: app/src/main/java/com/notonly/calendar/base/BaseActivity.java
package com.notonly.calendar.base;

import android.content.Context;
import android.os.Bundle;

import com.notonly.calendar.base.manager.AppManager;
import com.umeng.analytics.MobclickAgent;
import com.wangzhen.commons.toolbar.ToolbarActivity;

/**
 * Activity基类
 * Created by wangzhen on 16/2/23.
 */
public class BaseActivity extends ToolbarActivity {

    public Context mContext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AppManager.get().addActivity(this);
        mContext = this;
    }

    @Override
    protected void onResume() {
        super.onResume();
        MobclickAgent.onResume(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        MobclickAgent.onPause(this);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        AppManager.get().removeActivity(this);
    }

}

File: app/src/main/java/com/notonly/calendar/base/BaseActivity.kt
package com.notonly.calendar.base

import android.os.Bundle
import com.notonly.calendar.base.manager.AppManager
import com.umeng.analytics.MobclickAgent
import com.wangzhen.commons.toolbar.ToolbarActivity

/**
 * Activity基类
 * Created by wangzhen on 16/2/23.
 */
open class BaseActivity : ToolbarActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppManager.get().addActivity(this)
    }

    override fun onResume() {
        super.onResume()
        MobclickAgent.onResume(this)
    }

    override fun onPause() {
        super.onPause()
        MobclickAgent.onPause(this)
    }

    public override fun onDestroy() {
        super.onDestroy()
        AppManager.get().removeActivity(this)
    }
}